### DISCUSSION TITLE
How to handle dynamic forms?

### DISCUSSION THREAD

**Discussion by justintoman** [edmundhung/conform#273] [Q&A] [answered]
Let's say I have a checkbox "Other: Provide reason" and when checked, a text field becomes enabled and the user must enter a reason into it. And when the checkbox is unchecked I want to clear out the reason text field since it only applies when the checkbox is checked, so I don't want its value being included in the form submission.

Another scenario would be a user picks a country on an address form and then different fields appear based on the locale, i.e. US would have state and zip code, but other countries might have different things like districts, departments, prefectures, etc... so I need a dynamic set of fields to appear as the user is entering data.

I've used react-hook-form for a little while and I'd use `watch` for this kind of thing, but I don't know what the conform analog is.

My guess would be that I need to make a `useState` for each of the values I need to watch, but I was curious if there's a conform-specific way like RHF has.

Thanks!

**Comment by edmundhung** [comment, marked as answer]
Conform does not manage the form data as RHF does and it validates simply by looping over each entry on the FormData object. So what you said is pretty much correct: Just manage them with a simple `useState` and call it a day :)

**Reply by justintoman** [reply]
It seems like these questions come up a lot. Would you be open to adding a part on the docs that basically says what we discussed here? I would be happy to do it.
